Fix #258 enable mapping when not gcp kubernetes

tkmsaaaam opened the pull request:
Resolves #258

dashpole commented:
/gcbrun

psx95 reviewed (changes requested):
Thank you for the PR!

Overall looks good, I added some comments for improving code readability.

P.S: For fixing code style, you can run `./gradlew spotlessApply` from the root of the repository.

tkmsaaaam commented on shared/resourcemapping/src/main/java/com/google/cloud/opentelemetry/resource/ResourceTranslator.java:
```diff
@@ -153,13 +153,22 @@ public static GcpResource mapResource(Resource resource) {
       case ResourceAttributes.CloudPlatformValues.GCP_APP_ENGINE:
         return mapBase(resource, "gae_instance", GOOGLE_CLOUD_APP_ENGINE_INSTANCE_LABELS);
       default:
+        if (resource.getAttribute(ResourceAttributes.K8S_CLUSTER_NAME) != null) {
```
Resolves. https://github.com/GoogleCloudPlatform/opentelemetry-operations-java/pull/340/commits/cdb5691a46c1407e08c2ba0323547de16838104a

tkmsaaaam commented on shared/resourcemapping/src/main/java/com/google/cloud/opentelemetry/resource/ResourceTranslator.java:
```diff
@@ -153,13 +153,22 @@ public static GcpResource mapResource(Resource resource) {
       default:
+        if (resource.getAttribute(ResourceAttributes.K8S_CLUSTER_NAME) != null) {
+          if (resource.getAttribute(ResourceAttributes.K8S_CONTAINER_NAME) != null) {
+            return mapBase(resource, "k8s_container", K8S_CONTAINER_LABELS);
```
Resolves. https://github.com/GoogleCloudPlatform/opentelemetry-operations-java/pull/340/commits/18012d06a4ea1171a6ab309fbbc4dac8c99d49f7

tkmsaaaam commented on shared/resourcemapping/src/test/java/com/google/cloud/opentelemetry/resource/ResourceTranslatorTest.java:
Resolves https://github.com/GoogleCloudPlatform/opentelemetry-operations-java/pull/340/commits/772d6ecba70df881dc4d3bb7629dd1b042e6a617

tkmsaaaam replied:
But `./gradlew test` is failed. (Of course this test is succeeded)
Some `GCPResourceTest` are failed. I guess it has version's problem. So I will try to fix it by myself. At least I paste a part of error. If you know this problem well, please advise me.

```
GCPResourceTest > testUnknownPlatformResourceAttributesMapping() FAILED
    org.mockito.exceptions.base.MockitoException:
    Mockito cannot mock this class: class com.google.cloud.opentelemetry.detection.GCPPlatformDetector.

    If you're not sure why you're getting this error, please report to the mailing list.


    Java               : 17
    JVM vendor name    : Amazon.com Inc.
    JVM vendor version : 17.0.6+10-LTS
    JVM name           : OpenJDK 64-Bit Server VM
    JVM version        : 17.0.6+10-LTS
    JVM info           : mixed mode, sharing
    OS name            : Mac OS X
    OS version         : 14.4.1


    You are seeing this disclaimer because Mockito is configured to create inlined mocks.
    You can learn about inline mocks and their limitations under item #39 of the Mockito class javadoc.

    Underlying exception : org.mockito.exceptions.base.MockitoException: Could not modify all classes [class java.lang.Object, class com.google.cloud.opentelemetry.detection.GCPPlatformDetector]
        at app//com.google.cloud.opentelemetry.detectors.GCPResourceTest.testUnknownPlatformResourceAttributesMapping(GCPResourceTest.java:376)

        Caused by:
        org.mockito.exceptions.base.MockitoException: Could not modify all classes [class java.lang.Object, class com.google.cloud.opentelemetry.detection.GCPPlatformDetector]
            at app//net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:152)
            at app//net.bytebuddy.TypeCache$WithInlineExpunction.findOrInsert(TypeCache.java:365)
            at app//net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:174)
            at app//net.bytebuddy.TypeCache$WithInlineExpunction.findOrInsert(TypeCache.java:376)
            ... 1 more

            Caused by:
            java.lang.IllegalStateException:
            Byte Buddy could not instrument all classes within the mock's type hierarchy

            This problem should never occur for javac-compiled classes. This problem has been observed for classes that are:
             - Compiled by older versions of scalac
             - Classes that are part of the Android distribution
                at org.mockito.internal.creation.bytebuddy.InlineBytecodeGenerator.triggerRetransformation(InlineBytecodeGenerator.java:264)
                at org.mockito.internal.creation.bytebuddy.InlineBytecodeGenerator.mockClass(InlineBytecodeGenerator.java:211)
                at org.mockito.internal.creation.bytebuddy.TypeCachingBytecodeGenerator$1.call(TypeCachingBytecodeGenerator.java:46)
                at org.mockito.internal.creation.bytebuddy.TypeCachingBytecodeGenerator$1.call(TypeCachingBytecodeGenerator.java:43)
                at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:152)
                at net.bytebuddy.TypeCache$WithInlineExpunction.findOrInsert(TypeCache.java:365)
                at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:174)
                at net.bytebuddy.TypeCache$WithInlineExpunction.findOrInsert(TypeCache.java:376)
                at org.mockito.internal.creation.bytebuddy.TypeCachingBytecodeGenerator.mockClass(TypeCachingBytecodeGenerator.java:36)
                at org.mockito.internal.creation.bytebuddy.InlineByteBuddyMockMaker.createMockType(InlineByteBuddyMockMaker.java:379)
                at org.mockito.internal.creation.bytebuddy.InlineByteBuddyMockMaker.doCreateMock(InlineByteBuddyMockMaker.java:339)
                at org.mockito.internal.creation.bytebuddy.InlineByteBuddyMockMaker.createMock(InlineByteBuddyMockMaker.java:318)
                at org.mockito.internal.util.MockUtil.createMock(MockUtil.java:52)
                at org.mockito.internal.MockitoCore.mock(MockitoCore.java:61)
                at org.mockito.Mockito.mock(Mockito.java:1949)
                at org.mockito.Mockito.mock(Mockito.java:1860)
                ... 1 more

                Caused by:
                java.lang.IllegalArgumentException: Unsupported class file major version 61
                    at net.bytebuddy.jar.asm.ClassReader.<init>(ClassReader.java:196)
                    at net.bytebuddy.jar.asm.ClassReader.<init>(ClassReader.java:177)
                    at net.bytebuddy.jar.asm.ClassReader.<init>(ClassReader.java:163)
                    at net.bytebuddy.utility.OpenedClassReader.of(OpenedClassReader.java:86)
                    at net.bytebuddy.dynamic.scaffold.TypeWriter$Default$ForInlining.create(TypeWriter.java:3824)
                    at net.bytebuddy.dynamic.scaffold.TypeWriter$Default.make(TypeWriter.java:2166)
                    at net.bytebuddy.dynamic.scaffold.inline.RedefinitionDynamicTypeBuilder.make(RedefinitionDynamicTypeBuilder.java:224)
                    at net.bytebuddy.dynamic.scaffold.inline.AbstractInliningDynamicTypeBuilder.make(AbstractInliningDynamicTypeBuilder.java:123)
                    at net.bytebuddy.dynamic.DynamicType$Builder$AbstractBase.make(DynamicType.java:3595)
                    at org.mockito.internal.creation.bytebuddy.InlineBytecodeGenerator.transform(InlineBytecodeGenerator.java:375)
                    at java.instrument/java.lang.instrument.ClassFileTransformer.transform(ClassFileTransformer.java:244)
                    at java.instrument/sun.instrument.TransformerManager.transform(TransformerManager.java:188)
                    at java.instrument/sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:541)
                    at java.instrument/sun.instrument.InstrumentationImpl.retransformClasses0(Native Method)
                    at java.instrument/sun.instrument.InstrumentationImpl.retransformClasses(InstrumentationImpl.java:169)
                    at org.mockito.internal.creation.bytebuddy.InlineBytecodeGenerator.triggerRetransformation(InlineBytecodeGenerator.java:260)
                    ... 16 more

```

psx95 replied:
I think this issue is happening because you are using Java 17 to compile and run. It might be because we are using a version of Mockito which is not supported with Java 17.

If you have Java 11 installed on your machine, try building using that. It should work then. I don't think any of your recent commits have caused this error, but I wanted to confirm if this was happening before?

tkmsaaaam replied:
Thank you. I confirmed all tests are passed when Java 11.

psx95 reviewed (changes requested):
Thanks for making the changes! 

I opened a PR against your repository, with some code refactors in the test file to make it more consistent with our repository. PTAL and merge the changes from that PR in your branch.

PR: https://github.com/tkmsaaaam/opentelemetry-operations-java/pull/1

psx95 reviewed (approved):
LGTM

psx95 commented:
/gcbrun

psx95 commented:
@tkmsaaaam it looks like one of the file has some formatting issues, can you run `./gradlew spotlessApply` to confirm and commit any suggested changes ?

tkmsaaaam commented:
@psx95 Thank you for reviewing many times. I fixed it.

tkmsaaaam commented:
@psx95 My apologize. I fixed this issue. But I noticed it contained bug. So I fixed it. Please review it.
https://github.com/GoogleCloudPlatform/opentelemetry-operations-java/pull/342
